test/integration/backward_compatible/sql/JetTest.js

```diff
@@ -0,0 +1,198 @@
+const { Client } = require('../../../../');
+
+const chai = require('chai');
+
+chai.should();
+
+describe('Jet Test', function () {
+    let client;
+    let cluster;
+    let map;
+    let map2;
+
+    const mapName = 'a';
+    const mapName2 = 'b';
+    const jetEnabledConfig = `<?xml version="1.0" encoding="UTF-8"?>
+        <hazelcast xmlns="http://www.hazelcast.com/schema/config"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://www.hazelcast.com/schema/config
+            http://www.hazelcast.com/schema/config/hazelcast-config-4.0.xsd">
+            <jet enabled="true"></jet>
+        </hazelcast>`;
+
+    before(async function () {
+        TestUtil.markClientVersionAtLeast(this, '5.0');
+        TestUtil.markServerVersionAtLeast(this, client, '5.0');
+    });
+
+    beforeEach(async function () {
+        map = undefined;
+        map2 = undefined;
+        cluster = await RC.createCluster(null, jetEnabledConfig);
+        await RC.startMember(cluster.id);
+        client = await Client.newHazelcastClient({
+            clusterName: cluster.id
+        });
+    });
+
+    afterEach(async function () {
+        if (map) {
+            await map.destroy();
+        }
+        if (map2) {
+            await map2.destroy();
+        }
+        await client.shutdown();
+        await RC.terminateCluster(cluster.id);
+    });
+
+    it('should be able to run a generate_series query', async function () {
+        const result = client.getSql().execute('SELECT * FROM TABLE(generate_series(1,3))');
+        let counter = 0;
+        // eslint-disable-next-line no-unused-vars
+        for await (const row of result) {
+            counter++;
+        }
+        counter.should.be.eq(3);
+    });
+
+    it('should be able to run an infinite query', async function () {
+        const result = client.getSql().execute('SELECT * FROM TABLE(generate_stream(500))');
+        let counter = 0;
+
+        // eslint-disable-next-line no-unused-vars
+        for await (const row of result) {
+            counter++;
+            if (counter > 500) {
+                break;
+            }
+        }
+    });
+
+    it('should be able to run join query', async function () {
+        map = await client.getMap(mapName);
+        map2 = await client.getMap(mapName2);
+
+        const result = client.getSql().execute(`
+            CREATE MAPPING ${mapName} (__key DOUBLE, age INTEGER, name VARCHAR) TYPE IMap OPTIONS (
+              'keyFormat'='double',
+              'valueFormat'='json')
+        `);
+
+        await result.executeDeferred.promise; // wait for execution to end
+
+        const result2 = client.getSql().execute(`
+            CREATE MAPPING ${mapName2} (__key DOUBLE, name VARCHAR, height DOUBLE) TYPE IMap OPTIONS (
+              'keyFormat'='double',
+              'valueFormat'='json')
+        `);
+
+        await result2.executeDeferred.promise; // wait for execution to end
```

Comment:
maybe instead of this private API, await something `result.updateCount()`?

Reply:
Yeah it can be done and I am going to do that. this was my initial point of mentioning a new API that can be used to wait result execution. I think awaiting updateCount or other things are not very intuitive